src/main/java/org/folio/circulation/support/http/client/OkapiHttpClient.java

```diff
@@ -1,99 +1,43 @@
 package org.folio.circulation.support.http.client;
 
-import static org.folio.circulation.support.http.OkapiHeader.OKAPI_URL;
-import static org.folio.circulation.support.http.OkapiHeader.REQUEST_ID;
-import static org.folio.circulation.support.http.OkapiHeader.TENANT;
-import static org.folio.circulation.support.http.OkapiHeader.TOKEN;
-import static org.folio.circulation.support.http.OkapiHeader.USER_ID;
-
-import java.lang.invoke.MethodHandles;
 import java.net.URL;
-import java.util.function.Consumer;
-
-import org.apache.commons.lang.StringUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import io.vertx.core.Handler;
-import io.vertx.core.http.HttpClient;
-import io.vertx.core.http.HttpClientRequest;
-import io.vertx.core.http.HttpClientResponse;
-import io.vertx.core.json.Json;
-
-public class OkapiHttpClient {
-  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
-
-  private final HttpClient client;
-  private final URL okapiUrl;
-  private final String tenantId;
-  private final String token;
-  private final String userId;
-  private final String requestId;
-  private final Consumer<Throwable> exceptionHandler;
-
-  public OkapiHttpClient(HttpClient httpClient, URL okapiUrl, String tenantId,
-    String token, String userId, String requestId,
-    Consumer<Throwable> exceptionHandler) {
-
-    this.client = httpClient;
-    this.okapiUrl = okapiUrl;
-    this.tenantId = tenantId;
-    this.token = token;
-    this.userId = userId;
-    this.requestId = requestId;
-    this.exceptionHandler = exceptionHandler;
-  }
-
-  public VertxWebClientOkapiHttpClient toWebClient() {
-    return VertxWebClientOkapiHttpClient.createClientUsing(client, okapiUrl,
-      tenantId, token, userId, requestId);
-  }
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
 
-  public void put(URL url, Object body,
-    Handler<HttpClientResponse> responseHandler) {
+import org.folio.circulation.support.Result;
 
-    put(url.toString(), body, responseHandler);
-  }
+import io.vertx.core.json.JsonObject;
 
-  public void put(String url, Object body,
-    Handler<HttpClientResponse> responseHandler) {
+public interface OkapiHttpClient {
```

Comment:
Do you mind to add shorthand methods for frequently used operations:
* getById(String baseUrl, String id);
* putById(String baseUrl, String id, JsonObject body);
* deleteById(String baseUrl, String id);

Reply:
These higher level methods should be covered by the `CollectionResourceClient`. The `OkapiHttpClient` is intended to offer the most basic operations, and most often not used directly.

Are methods like these present on the `CollectionResourceClient`?

Reply:
Yes, they are. Resolving conversation then